designs/exportable-subflow/README.md

```diff
@@ -24,28 +24,52 @@ are in the roadmap. (*TODO: add a link to the design doc once added*).
+By clicking **Export SUBFLOW** button on SUBFLOW template setting panel, SUBFLOW export panel is shown.
 
-New node created from SUBFLOW is sealed.  Thus, its internal flow or UI definition can not be edited.
+In this panel, new node information such as node name, version, etc. can be specified. There is **Include All SUBFLOW** check box on this panel.  If this is checked, SUBFLOWs used within the exported SUBFLOW are recursively included in exported SUBFLOW Module.  If **Editable** check box on this panel is checked, SUBFLOW template can be edited on imported Node-RED editor.  Otherwise, the imported SUBLOW template is sealed. Thus, its internal flow or UI definition can not be edited.
 
-#### Importing SUBFLOW Node
+#### Importing SUBFLOW Module
 
-Node represented as JSON data can be imported in Node-RED environment from palette management UI.
+Node represented as JSON data can be imported in Node-RED environment from import new node panel.
 
 ![Subflow-import-UI](Subflow-import-UI.png)
 
-There is **Install(JSON)** button on top of the panel.  After importing JSON, new node is installed on editor palette.
+#### Uninstalling SUBFLOW Module
+
+SUBFLOW Module are listed in *Manage Palette > Palette > Nodes* tab.  Similar to normal nodes, *SUBFLOW Module*s listed in this tab will have **remove** button.  If a SUBFLOW Module includes other SUBFLOW Modules and is sealed, contained SUBFLOW Modules are also deleted.
+
+#### SUBFLOW Module Format
+
+Exported SUBFLOW Module is represented by JSON array format which is similar to current flow representation.  
+
+Following is a list of properties exported SUBFLOW node in SUBFLOW Module may contain (+ means that property is optional):
+
+|      | name    | type   | description                        |
+| ---- | ------- | ------ | ---------------------------------- |
+| 1    | flow    | array  | array of contained nodes in SUBFLOW |
+| 2    | sealed+ | bool   | SUBFLOW template is not editable   |
+| 3    | hidden+ | bool   | SUBFLOW template is hidden         |
+| 4    | moduleName | string | name of SUBFLOW module             |
+| 5    | color+  | string | color of SUBFLOW                   |
+| 6    | version | string | version of SUBFLOW (format: x.y.z) |
+| 7    | author+ | string | author of SUBFLOW                  |
+| 8    | license+ | string | license of SUBFLOW                 |
+| 9    | keywords+ | array  | array of keywords of SUBFLOW       |
+| 10   | description+ | string | simple description of SUBFLOW      |
+
+Exported SUBFLOW Module can be **sealed** which means that internal of the SUBFLOW Module is hidden from users.  
+
+`flow` property is an array of nodes that are used in its SUBFLOW template.  *id* of nodes in SUBFLOW module must be *name*-*ID* where *ID* is a original *id* of the node.
```

Comment:
Where has this requirement for the ids of the nodes inside the subflow come from? What does it solve doing this?

Reply:
Not discussed this issue on teleconference.
This is for different flows that use same SUBFLOW Template.
If ID is random string, existing flow can not refer to imported SUBFLOW Template.